plugins/inventory/docker.py

```diff
@@ -0,0 +1,321 @@
+            - Default is to use SSH (C(ssh)). For this, the options I(default_ip) and
+              I(private_ssh_port) are used.
+            - Alternatively, C(docker-cli) selects the
+              R(docker connection plugin,ansible_collections.community.docker.docker_connection),
+              and C(docker-api) selects the
+              R(docker_api connection plugin,ansible_collections.community.docker.docker_api_connection).
+        type: str
+        default: docker-api
+        choices:
+            - ssh
+            - docker-cli
+            - docker-api
+
+    verbose_output:
+        description:
+            - Toggle to (not) include all available inspection metadata.
+            - Note that all top-level keys will be transformed to the format C(docker_xxx).
+              For example, C(HostConfig) is converted to C(docker_hostconfig).
+            - If this is C(false), these values can only be used during I(constructed), I(groups), and I(keyed_groups).
+            - The C(docker) inventory script always added these variables, so for compatibility set this to C(true).
+        type: bool
+        default: false
+
+    default_ip:
+        description:
+            - The IP address to assign to ansible_host when the container's SSH port is mapped to interface
+              '0.0.0.0'.
+            - Only used if I(connection_type) is C(ssh).
+        type: str
+        default: 127.0.0.1
+
+    private_ssh_port:
+        description:
+            - The port containers use for SSH.
+            - Only used if I(connection_type) is C(ssh).
+        type: int
+        default: 22
+
+    add_legacy_groups:
+        description:
+            - "Add the same groups as the C(docker) inventory script does. These are the following:"
+            - "C(<container id>): contains the container of this ID."
+            - "C(<container name>): contains the container that has this name."
+            - "C(<container short id>): contains the containers that have this short ID (first 13 letters of ID)."
+            - "C(image_<image name>): contains the containers that have the image C(<image name>)."
+            - "C(stack_<stack name>): contains the containers that belong to the stack C(<stack name>)."
+            - "C(service_<service name>): contains the containers that belong to the service C(<service name>)"
+            - "C(<docker_host>): contains the containers which belong to the Docker daemon I(docker_host).
+              Useful if you run this plugin against multiple Docker daemons."
+            - "C(running): contains all containers that are running."
+            - "C(stopped): contains all containers that are not running."
+            - If this is not set to C(true), you should use keyed groups to add the containers to groups.
+              See the examples for how to do that.
+        type: bool
+        default: false
+'''
+
+EXAMPLES = '''
+# Minimal example using local Docker daemon
+plugin: community.docker.docker
```

Comment:
I think the `docker` is a misleading name. This inventory plugin is for containers so in my opinion it should be named as `docker_container` or `docker_containers` which will match the naming convention used for the other two plugins.

Reply:
Good idea! I kept the name similar to the old inventory plugin, but there's really no need for that. I think `docker_containers` sounds good. Actually, we could also call it just `containers`, since the FQCN `community.docker.containers` would already include "docker".

Reply:
I've now named it `docker_containers` for consistency.